scrunch/mutable_dataset.py

```diff
@@ -113,7 +115,8 @@ def join(self, left_var, right_ds, right_var, columns=None,
         progress = self.resource.variables.post(payload)
         # poll for progress to finish or return the url to progress
         if wait:
```

Comment:
we should change this to `wait or timeout` in case they stop using `wait` and start using `timeout` as the preferred argument. Having two arguments that do something similar will become confusing.